draft-ietf-masque-connect-ip.md

```diff
@@ -280,35 +289,47 @@ address, the receiver of this capsule is allowed to send packets from any source
 address that falls within the prefix.
 {: spacing="compact"}
 
-If an endpoint receives multiple ADDRESS_ASSIGN capsules, all of the assigned
-addresses or prefixes can be used. For example, multiple ADDRESS_ASSIGN capsules
-are necessary to assign both IPv4 and IPv6 addresses.
+If an endpoint receives multiple Assigned Addresses (in one or more capsules),
+all of the assigned addresses or prefixes can be used. For example, multiple
+Assigned Addresses are necessary to assign both IPv4 and IPv6 addresses.
 
 In some deployments of CONNECT-IP, an endpoint needs to be assigned an address
 by its peer before it knows what source address to set on its own packets. For
 example, in the Remote Access case ({{example-remote}}) the client cannot send
-IP packets until it knows what address to use. In these deployments, endpoints
-need to send ADDRESS_ASSIGN capsules to allow their peers to send traffic.
+IP packets until it knows what address to use. In these deployments, the
+endpoint that is expecting an address assignment MUST send an ADDRESS_ASSIGN
+capsule. This isn't required if the endpoint does not need any address
+assignment, for example when it is configured out of band with static addresses.
+
+While ADDRESS_ASSIGN capsules are commonly sent in response to ADDRESS_REQUEST
+capsules, endpoints MAY send ADDRESS_ASSIGN capsules unprompted.
```

Comment:
Do we actually want this text? I agree that this is permitted per our discussion as "send an ADDRESS_REQUEST if you need an ADDRESS_ASSIGN" does not preclude unsolicited ADDRESS_ASSIGN, but I can't think of a use case here that requires this clarification.

Reply:
I think it's nice to have a MAY here just to make sure people don't read it as a MUST NOT.